samples/msal-node-samples/ElectronSystemBrowserTestApp/src/AuthProvider.ts

```diff
@@ -0,0 +1,237 @@
+
+    async getTokenSilent(
+        tokenRequest: SilentFlowRequest
+    ): Promise<AuthenticationResult> {
+        try {
+            return await this.clientApplication.acquireTokenSilent(
+                tokenRequest
+            );
+        } 
+        
+        catch (error) {
+            console.log(
+                "Silent token acquisition failed, acquiring token using pop up"
+            );
+            
+            await this.login();
+            return null
+            
+        }
+    }
+
+
+    async getAuthCode(tokenRequest: AuthorizationUrlRequest): Promise<string> {
+        // Generate PKCE Challenge and Verifier before request
+        const cryptoProvider = new CryptoProvider();
+        const { challenge, verifier } =
+            await cryptoProvider.generatePkceCodes();
+
+        this.challenge = challenge;
+        this.verifier = verifier;
+
+        // Add PKCE params to Auth Code URL request
+        const authCodeUrlParams = {
+            ...this.authCodeUrlParams,
+            scopes: tokenRequest.scopes,
+            codeChallenge: this.challenge,
+            codeChallengeMethod: "S256",
+        };
+        try {
+            const authCodeUrl = await this.clientApplication.getAuthCodeUrl(
+                authCodeUrlParams
+            );
+            return authCodeUrl;
+        } catch (error) {
+            console.log(error);
+            throw error;
+        }
+    }
+
+    async login(): Promise<void> {
+        const getAuthURL = await this.getAuthCode(this.authCodeUrlParams);
+        Opener(getAuthURL);
+    }
+
+    async getTokenByCode(url: string): Promise<AccountInfo>{
+        try {
+            const parsedUtl = urlparse(url, true);
+             let code = parsedUtl.query.code;
+             const authResult = await this.clientApplication.acquireTokenByCode(
+                 {
+                     ...this.authCodeRequest,
+                     code: code,
+                     codeVerifier: this.verifier,
+                 }
+             );
+            return this.handleResponse(authResult);
+        } catch (error) {
+            throw error
+        }
+    }
+
+    async loginSilent(): Promise<AccountInfo> {
+        if (!this.account) {
+            this.account = await this.getAccount();
+        }
+
+        return this.account;
+    }
+
+    async logout(): Promise<void> {
+        if (this.account) {
+            await this.clientApplication
+                .getTokenCache()
+                .removeAccount(this.account);
+            this.account = null;
+        }
+    }
+
+
+    /**
+     * Handles the response from a popup or redirect. If response is null, will check if we have any accounts and attempt to sign in.
+     * @param response
+     */
+    private async handleResponse(response: AuthenticationResult) {
+        if (response !== null) {
+            this.account = response.account;
+        } else {
+            this.account = await this.getAccount();
+        }
```

Comment:
Could you do the following?:

this.account = response?.account || await this.getAccount();

Reply:
Updated!